### PR TITLE
Create org.remmina.Remmina

### PR DESCRIPTION
**Pull request by amtlib-dot-dll**
cc @larchunix

Copied from https://github.com/FreeRDP/Remmina/tree/next/flatpak. I am not the orginal author. I just want to use the `flatpak` version without compile it by myself :laughing:

### CODE REVIEW

File: org.remmina.Remmina.json [hunk 1]
```diff
@@ -0,0 +1,494 @@
...
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.26",
+    "sdk": "org.gnome.Sdk",
+    "build-options" : {
+        "cflags": "-O2 -g",
+        "cxxflags": "-O2 -g"
+    },
+    "command": "remmina",
+    "rename-icon": "remmina",
+    "desktop-file-name-prefix": "(Flatpak) ",
```
Comment (TingPing) [inline org.remmina.Remmina.json]: Unnecessary

File: org.remmina.Remmina.json [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+        "/build-aux",
+        "/include",
+        "/lib/cmake",
+        "/lib/pkgconfig",
+        "/share/gtk-doc",
+        "/share/man",
+        "/share/pkgconfig"
+    ],
+    "finish-args": [
+        "--own-name=org.remmina.Remmina",
```
Comment (TingPing) [inline org.remmina.Remmina.json]: Not needed

File: org.remmina.Remmina.json [hunk 3]
```diff
@@ -0,0 +1,494 @@
...
+        /* File transfer (although limited to user directory) */
+        "--filesystem=home",
+        /* Desktop notifications */
+        "--talk-name=org.freedesktop.Notifications",
+        /* Needed to save login credentials */
+        "--talk-name=org.freedesktop.secrets",
+        /* Appindicator */
+        "--talk-name=org.kde.StatusNotifierWatcher",
+        /* Query GNOME Shell version (to display systray icon if supported) */
+        "--talk-name=org.gnome.Shell",
```
Comment (TingPing) [inline org.remmina.Remmina.json]: This is an upstream bug, but this is completely the wrong way to handle this.

File: org.remmina.Remmina.json [hunk 4]
```diff
@@ -0,0 +1,494 @@
...
+                    "url": "https://xorg.freedesktop.org/releases/individual/lib/libxkbfile-1.0.9.tar.bz2",
+                    "sha256": "51817e0530961975d9513b773960b4edd275f7d5c72293d5a151ed4f42aeb16a"
+                }
+            ]
+        },
+        {
+            "name": "nxproxy",
+            "buildsystem": "simple",
+            "build-commands": [
+                "make CONFIGURE='./configure --prefix=/app'",
```
Comment (TingPing) [inline org.remmina.Remmina.json]: I don't know why this is the `simple` build system.

File: org.remmina.Remmina.json [hunk 5]
```diff
@@ -0,0 +1,494 @@
...
+                {
+                    "type": "archive",
+                    "url": "https://code.x2go.org/releases/source/nx-libs/nx-libs-3.5.0.33-lite.tar.gz",
+                    "sha256": "e43824f820f8d307207aa4686c68f9b0808909a6b6cd2c7d75b751b14dd3129f"
+                }
+            ]
+        },
+        {
+            "name": "freerdp",
+            "buildsystem": "cmake",
```
Comment (TingPing) [inline org.remmina.Remmina.json]: `cmake-ninja`

File: org.remmina.Remmina.json [hunk 6]
```diff
@@ -0,0 +1,494 @@
...
+                            "url": "https://pypi.python.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl",
+                            "sha256": "832dc0e10feb1aa2c68dcc57dbb658f1c7e65b9b61af69048abc87a2db00a0eb"
+                        }
+                    ]
+                },
+                {
+                    "name": "lz4",
+                    "buildsystem": "simple",
+                    "build-commands": [
+                        "make -C lib lib",
```
Comment (TingPing) [inline org.remmina.Remmina.json]: No need for simple build system:

```json
{
  "subdir": "lib",
  "no-autogen": true,
  "make-args": ["PREFIX=/app", "lib"],
}
```

File: org.remmina.Remmina.json [hunk 7]
```diff
@@ -0,0 +1,494 @@
...
+                            "url": "https://www.spice-space.org/download/releases/spice-protocol-0.12.13.tar.bz2",
+                            "sha256": "89ee11b202d2268e061788e6ace114e1ff18c7620ae64d1ca3aba252ee7c9933"
+                        }
+                    ]
+                }
+            ]
+        },
+        {
+            "name": "libssh",
+            "buildsystem": "cmake",
```
Comment (TingPing) [inline org.remmina.Remmina.json]: `cmake-ninja`

File: org.remmina.Remmina.json [hunk 8]
```diff
@@ -0,0 +1,494 @@
...
+                            "url": "https://xorg.freedesktop.org/releases/individual/app/xkbcomp-1.4.0.tar.bz2",
+                            "sha256": "bc69c8748c03c5ad9afdc8dff9db11994dd871b614c65f8940516da6bf61ce6b"
+                        }
+                    ]
+                }
+            ]
+        },
+        {
+            "name": "remmina",
+            "buildsystem": "cmake",
```
Comment (TingPing) [inline org.remmina.Remmina.json]: `cmake-ninja`

File: org.remmina.Remmina.json [hunk 9]
```diff
@@ -0,0 +1,494 @@
...
+            "config-opts": [
+                "-DCMAKE_INSTALL_LIBDIR:PATH=/app/lib",
```
Comment (TingPing) [inline org.remmina.Remmina.json]: This can be removed.

File: org.remmina.Remmina.json [hunk 10]
```diff
@@ -0,0 +1,494 @@
...
+                            "sha256": "bc69c8748c03c5ad9afdc8dff9db11994dd871b614c65f8940516da6bf61ce6b"
+                        }
+                    ]
+                }
+            ]
+        },
+        {
+            "name": "remmina",
+            "buildsystem": "cmake",
+            "config-opts": [
```
Comment (TingPing) [inline org.remmina.Remmina.json]: `-DCMAKE_BUILD_TYPE=Release`

File: org.remmina.Remmina.json (hunk 5 above)
Comment (amtlib-dot-dll) [reply org.remmina.Remmina.json]: Thank you for your review, by the way, what is the difference between `cmake` and `cmake-ninja`?

Comment (TingPing) [reply org.remmina.Remmina.json]: `cmake` generates Makefiles, `ninja` is simply faster than `make` so when it works use it.

Comment (amtlib-dot-dll) [reply org.remmina.Remmina.json]: Alright :man_facepalming:

**Comment by nedrichards** [comment]
bot, build org.remmina.Remmina

**Comment by nedrichards** [comment]
Builds fail with appdata not available https://flathub.org/builds/#/builders/2/builds/221

**Comment by larchunix** [comment]
@amtlib-dot-dll Thanks for pushing my flatpak manifest here.

I opened a PR on Remmina to fix most of the issues raised by @TingPing (https://github.com/FreeRDP/Remmina/pull/1526).

The error raised by @nedrichards has been fixed upstream by https://github.com/FreeRDP/Remmina/commit/a860ca270328f545e4934dfe3c0b78433a446506.

**Comment by nedrichards** [comment]
@larchunix that's awesome! do you know when the next upstream release that includes that will be?

**Comment by antenore** [comment]
You guys are amazing! Thanks a lot!!!!!!!!

I don't know when we will release a new version, I've just merged these modifications upstream in our main branch (named next)

If you need a release to go ahead, we can do it I think.

What do you think @giox069 ?

**Comment by amtlib-dot-dll** [comment]
@larchunix @antenore @giox069

I have an idea. FreeRDP is well-known, and more famous than Remmina, so if we change the `app-id` from `org.remmina.Remmina` to `org.freerdp.Remmina`, people who use search engines to look for FreeRDP will know about Remmina, especially when they search for FreeRDP in the Flathub directory. What do you think?

**Comment by antenore** [comment]
Personally I have nothing to objects, I think it's a good idea, moreover we belong to the FreeRDP organisation.

**Comment by antenore** [comment]
Finally I'm not sure anymore. The Remmina dbus id is org.remmina.Remmina and it's better to stay in-line with the dbus name.

As soon as we will endorse flatpak, I'm sure we won't need advertising.

So far flatpack and the flathub community looks amazing :-)

Thanks for supporting us!

**Comment by nedrichards** [comment]
Searching is about the information in your appdata, not the app id. I
really wouldn't change that.

**Comment by antenore** [comment]
If it doesn't hurt we can change it

**Comment by amtlib-dot-dll** [comment]
Searching may also be something like this `flatpak remote-ls flathub | grep -i freerdp` @nedrichards

**Comment by TingPing** [comment]
The best app-id is a discussion to have with upstream as that is where these appdata files should go.

**Comment by TingPing** [comment]
> Searching may also be something like this flatpak remote-ls flathub | grep -i freerdp @nedrichards

Well that is just wrong: `flatpak search freerdp`.

**Comment by nedrichards** [comment]
bot, build org.remmina.Remmina

**Comment by antenore** [comment]
@nedrichards 
To resume..

1) Do you need other changes from our side?
2) Do you need an official release?

Thanks

**Comment by nedrichards** [comment]
@antenore a release is always helpful, but so long as this is all merged upstream you can attach a patch here since it's not a functional change or user visible bug fix. I don't want to cause you extra work that's unnecessary.

**Comment by antenore** [comment]
Thanks @nedrichards we are preparing a new release, but I don't know exactly when will be available.
I'm waiting with impatience to see Remmina in Flathub, so whatever you need I'm available without any issues.

**Comment by amtlib-dot-dll** [comment]
@TingPing 
> Well that is just wrong: `flatpak search freerdp`.

Sorry, but I tried and failed like this
```shell
$ flatpak search code
No matches found
$ flatpak search com.visualstudio.code
No matches found
$ flatpak search com.visualstudio.code.oss
No matches found
$ 
```
I installed `com.visualstudio.code.oss` locally. Is anything missing?

**Comment by barthalion** [comment]
Yes, most likely `flatpak update --appstream`.

**Comment by amtlib-dot-dll** [comment]
Any news guys? :grin:

**Comment by TingPing** [comment]
You didn't address my comment.

**Comment by amtlib-dot-dll** [comment]
@TingPing Sorry, but which one?🐒

File: org.remmina.Remmina.json (hunk 1 above)
Comment (TingPing) [reply org.remmina.Remmina.json]: @amtlib-dot-dll This one =)

Comment (amtlib-dot-dll) [reply org.remmina.Remmina.json]: https://github.com/amtlib-dot-dll/flathub/commit/aecd39712ea6e4c75acef535dcf37c2584da8598 Thank you for notifying :joy:

**Comment by barthalion** [comment]
bot, build org.remmina.Remmina

**Comment by amtlib-dot-dll** [comment]
Any more news guys? :grinning: :

**Comment by nedrichards** [comment]
I thought we were waiting for an upstream release or for the changes accepted upstream to be added as a patch to this version (until the next release comes out)

**Comment by antenore** [comment]
@nedrichards we prefer to don't release right now as we are still working hard for the next one.

It's not clear for me, sorry, what I'm supposed to do to accept the patches.
As soon as you'll tell me I'll do what is needed.
Thanks!

**Comment by nedrichards** [comment]
Here's an example of what I mean by a patch. https://github.com/flathub/org.pitivi.Pitivi/blob/master/org.pitivi.Pitivi.json#L138

Flatpak-builder can apply patch files to the source code it uses for each module. You can take the upstream commits, save them as patch files and use this sort of syntax to apply and test them.

**Comment by antenore** [comment]
@nedrichards I'd like to just propose our json file (from upstream) as the one to use for the flathub builds

https://github.com/FreeRDP/Remmina/blob/next/flatpak/org.remmina.Remmina.json 

As I don't have rights to modify this PR I can only submit a new one or attach a patch file of the json file.

Regarding the commit to use (https://github.com/FreeRDP/Remmina/blob/next/flatpak/org.remmina.Remmina.json#L481), in our json file we use directly the main branch (named 'next').

The alternatives are to use the releases, or specific commits as @amtlib-dot-dll has done in this PR.

Our (as mainstream devs) preferred choice would be to use 'next'. We do a lo of testing before to push the code in there... Moreover @larchunix recently has fixed some small annoyances.

If you want I can just submit a new PR with the updated flatpak based on your input, but I need to know exactly what you prefer.

In the (near) future, we will submit PRs to improve it further also based on the @TingPing feedback. 

Thanks!

**Comment by amtlib-dot-dll** [comment]
@antenore Hi, I sent you an invitation so you can modify this PR via [this branch](https://github.com/amtlib-dot-dll/flathub/tree/remmina), however I remembered that according to [App Requirements · flathub/flathub Wiki](https://github.com/flathub/flathub/wiki/App-Requirements#stable-releases-reproducible-builds)

> Manifests should therefore refer to tarballs or git tags and not the tip of a branch.

I am afraid that merely the main branch won't be accepted

**Comment by antenore** [comment]
@amtlib-dot-dll thanks! We're discussing internally to do a new release

**Comment by barthalion** [comment]
Some specific commit will also be fine as long as you are willing to support that in case of issues reported by upstream. Branch alone won't do as we want builds to be reproducible.

**Comment by nedrichards** [comment]
bot, build org.remmina.Remmina

**Comment by antenore** [comment]
We have found a couple of bugs that impact old distros, like Ubuntu 14.04.
We are working on these bugs ans we will re-tag a new commit or do a new release.

Sorry for the inconvenient.

**Comment by nedrichards** [comment]
No worries, power of upstream!

**Comment by amtlib-dot-dll** [comment]
New tag coming, thanks to @antenore and other upstream Remmina developers!

@nedrichards May you trigger another bot build? Thanks

**Comment by TingPing** [comment]
bot, build org.remmina.Remmina

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.remmina.Remmina

**Comment by nedrichards** [comment]
@antenore let us know the github usernames of any other upstream developers who we should add to the new repo.

**Comment by antenore** [comment]
@nedrichards Thanks!

Those in CC to this message

cc @giox069 @larchunix @amtlib-dot-dll

**Comment by TingPing** [comment]
@antenore Added.

**Comment by antenore** [comment]
@nedrichards @TingPing sorry to comment on a closed PR

How to ask for membership changes? 

I'd like to remove larchunix and add @myheroyuki 

Thanks!

**Comment by TingPing** [comment]
@antenore Done. An invitation was sent.

**Comment by antenore** [comment]
> 

Thank you @TingPing !!!

**Comment by myheroyuki** [comment]
@TingPing Thank you, just accepted

**Pull request closed without merging**